### PR TITLE
Dev stacked interp

### PR DESCRIPTION
**Pull request by InferiorBrain**
Git this one a try.

Thanks,

Phil

### CODE REVIEW

**Comment by dmbee** [comment]
Thanks Phil - please remove unrelated files from your pull request (eg those related to venv)

File: seglearn/transform.py [hunk 1]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
         return Xt, yt, swt
 
 
+class StackedInterp(BaseEstimator, XyTransformerMixin):
+    '''
+    Extension of Interp where input is in stacked format.
+    Transformer for resampling time series data to a fixed period over closed interval
```
Comment (dmbee) [inline seglearn/transform.py]: Here you should clearly describe the 'stacked' format. 
Specifically indicate it must be at least 3 columns (time, var_type, var_value)
Additional columns are treated as additional channels for the variable eg (time, var_type, var_value1, var_value2)

File: seglearn/transform.py [hunk 2]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
...
+            parameter.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        '''
+        self._check_data(X)
+
+        if not X[0].ndim > 1:
```
Comment (dmbee) [inline seglearn/transform.py]: > 2
also change comment

File: seglearn/transform.py [hunk 3]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
...
+        return self
+    
+    def _check_data(self, X):
+        '''
+        Checks that unique identifiers are consistent between time series.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_series, ...]
+        Time series data and (optionally) contextual data
```
Comment (dmbee) [inline seglearn/transform.py]: fix the docstring

File: seglearn/transform.py [hunk 4]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
...
+        N: scalar
+        Number of time series inputs
+        '''
+        
+        if len(X) > 1:
+            sval = np.unique(X[0][:, 1])
```
Comment (dmbee) [inline seglearn/transform.py]: you don't raise an error when the check fails

File: seglearn/transform.py (hunk 1 above)
Comment (dmbee) [reply seglearn/transform.py]: also indicate that each time series must include all the variables

File: seglearn/transform.py [hunk 5]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
...
+    def _interp(self, t_new, t, x, kind):
+        interpolator = interp1d(t, x, kind=kind, copy=False, bounds_error=False,
+                                fill_value="extrapolate", assume_sorted=True)
+        return interpolator(t_new)
+
+    def transform(self, X, y=None, sample_weight=None):
+        '''
+        Transforms the time series data with linear direct value interpolation
+        If y is a time series and passed, it will be transformed as well
+        The time dimension is removed from the data
```
Comment (dmbee) [inline seglearn/transform.py]: the time dimension and variable labels are removed

File: seglearn/transform.py [hunk 6]
```diff
@@ -629,6 +630,169 @@ def transform(self, X, y=None, sample_weight=None):
...
+        X_new : array-like, shape [n_series, ]
+            transformed time series data
+        y_new : array-like, shape [n_series]
+            expanded target vector
+        sample_weight_new : array-like or None
+            None is returned if target is changed. Otherwise it is returned unchanged.
+        '''
+        
+        xt, xc = get_ts_data_parts(X)
+        yt = y
```
Comment (dmbee) [inline seglearn/transform.py]: you don't check the data

File: examples/plot_stacked_interp.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+# Plot the 2 time series, each of the sensors versus the new sample period for 2 sample periods
+sample_periods = [(10. / 100.)*(inNanoseconds*10**9), (1. / 100.)*(inNanoseconds*10**9), (0.1 / 100.)*(inNanoseconds*10**9)]
+
+f, axarr = plt.subplots(2, 3)
+
+# Matplotlib style parameters
+left = 0.125  # the left side of the subplots of the figure
+right = 0.9  # the right side of the subplots of the figure
+bottom = 0.1  # the bottom of the subplots of the figure
+top = 0.9  # the top of the subplots of the figure
```
Comment (dmbee) [inline examples/plot_stacked_interp.py]: I would just base the example on the toy data.

**Comment by qtux** [comment]
Just a small notice: I suggest to squash this before merging to avoid adding these binaries to the git history (and some of the git commits which removed and added features not belonging to this feature branch). This would not unnecessarily bloat the repository and ease future development, e.g. when bisecting code.

**Comment by dmbee** [comment]
Thanks Matthias - will do. Much appreciated.

**Pull request closed without merging**